src/devices/device.ts

```diff
@@ -82,16 +97,102 @@ export class Device extends EventEmitter {
         this.hub.send(data, characteristic, callback);
     }
 
-    public subscribe (mode: number) {
+    protected sendWithMode (modeName: string, data: Buffer, characteristic: string = Consts.BLECharacteristic.LPF2_ALL, callback?: () => void) {
+        const previousMode = this.mode;
+        this.subscribe(modeName);
+        this.send(data, characteristic, () => {
+            if (previousMode) {
+                this.subscribe(previousMode);
+                if (callback) {
+                    callback();
+                }
+            }
+        });
+    }
+
+    protected sendLinearPowerCommand (value: number) {
+        return new Promise((resolve) => {
+            if (this.isWeDo2SmartHub) {
+                const data = Buffer.from([this.portId, 0x01, 0x02, value]);
+                this.send(data, Consts.BLECharacteristic.WEDO2_MOTOR_VALUE_WRITE);
+            } else {
+                const data = Buffer.from([0x81, this.portId, 0x11, 0x51, 0x00, value]);
```

Comment:
There can be multiple values - the LED light takes seperate R G B values, for instance.

Reply:
I did not know it was the same header everywhere, I planned to add the mapSpeed logic directly to this method by adding min/max parameters and maybe the 127 thing.

Reply:
Yeah, see here for usage when setting hub LED RGB color: https://github.com/nathankellenicki/node-poweredup/blob/feature/devices/src/devices/hubled.ts#L48

Here is the documentation for this command: https://lego.github.io/lego-ble-wireless-protocol-docs/index.html#writedirectmodedata